client/src/app/site/users/services/user-repository.service.ts

```diff
@@ -232,6 +234,26 @@ export class UserRepositoryService extends BaseRepository<ViewUser, User> {
         return results;
     }
 
+    /**
+     * Searches and returns Users by full name
+     *
+     * @param number: A participant number
+     * @returns all users matching that number
+     */
+    public getUsersByNumber(number: string): ViewUser[] {
```

Comment:
Whats about `return this.getViewModelList().filter(user => user.participant_number === number);`?